gobblin-metrics/src/main/java/gobblin/metrics/MetricContext.java

```diff
@@ -0,0 +1,665 @@
+   *
+   * <p>
+   *   This method will return fully-qualified metric names if the {@link MetricContext} is configured
+   *   to report fully-qualified metric names.
+   * </p>
+   */
+  @Override
+  public SortedMap<String, Histogram> getHistograms(MetricFilter filter) {
+    return this.reportFullyQualifiedNames ?
+        super.getHistograms(filter) : getSimplyNamedMetrics(Histogram.class, Optional.of(filter));
+  }
+
+  /**
+   * See {@link com.codahale.metrics.MetricRegistry#getMeters()}.
+   *
+   * <p>
+   *   This method will return fully-qualified metric names if the {@link MetricContext} is configured
+   *   to report fully-qualified metric names.
+   * </p>
+   */
+  @Override
+  public SortedMap<String, Meter> getMeters() {
+    return this.reportFullyQualifiedNames ?
+        super.getMeters() : getSimplyNamedMetrics(Meter.class, Optional.<MetricFilter>absent());
+  }
+
+  /**
+   * See {@link com.codahale.metrics.MetricRegistry#getMeters(com.codahale.metrics.MetricFilter)}.
+   *
+   * <p>
+   *   This method will return fully-qualified metric names if the {@link MetricContext} is configured
+   *   to report fully-qualified metric names.
+   * </p>
+   */
+  @Override
+  public SortedMap<String, Meter> getMeters(MetricFilter filter) {
+    return this.reportFullyQualifiedNames ?
+        super.getMeters(filter) : getSimplyNamedMetrics(Meter.class, Optional.of(filter));
+  }
+
+  /**
+   * See {@link com.codahale.metrics.MetricRegistry#getTimers()}.
+   *
+   * <p>
+   *   This method will return fully-qualified metric names if the {@link MetricContext} is configured
+   *   to report fully-qualified metric names.
+   * </p>
+   */
+  @Override
+  public SortedMap<String, Timer> getTimers() {
+    return this.reportFullyQualifiedNames ?
+        super.getTimers() : getSimplyNamedMetrics(Timer.class, Optional.<MetricFilter>absent());
+  }
+
+  /**
+   * See {@link com.codahale.metrics.MetricRegistry#getTimers(com.codahale.metrics.MetricFilter)}.
+   *
+   * <p>
+   *   This method will return fully-qualified metric names if the {@link MetricContext} is configured
+   *   to report fully-qualified metric names.
+   * </p>
+   */
+  @Override
+  public SortedMap<String, Timer> getTimers(MetricFilter filter) {
+    return this.reportFullyQualifiedNames ?
+        super.getTimers(filter) : getSimplyNamedMetrics(Timer.class, Optional.of(filter));
+  }
+
+  @Override
+  public Counter counter(String name) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public Meter meter(String name) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public Histogram histogram(String name) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public Timer timer(String name) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public <T extends Metric> T register(String name, T metric)
```

Comment:
Yes, I saw that, but if the user calls register explicitly, it will be unclear for the user whether that is also registering parent metrics in all parent contexts.